examples/2-handle-prompt/index.js

```diff
@@ -0,0 +1,100 @@
+import http from 'http';
+import { Octokit } from "@octokit/rest";
+import { 
+    createAckEvent, 
+    createDoneEvent, 
+    createTextEvent, 
+    parseRequestBody, 
+    prompt
+ } from "@copilot-extensions/preview-sdk";
+
+
+// Define the handler function
+const handler = async (request, response) => {
+    if (request.method === 'POST') {
+        console.log('Received POST request');
+
+        // get a token to use
+        const tokenForUser = request.headers['x-github-token'];
+        
+        // get the user information with the token        
+        const octokit = new Octokit({ auth: tokenForUser });
+        const user = await octokit.request("GET /user");
+        const userHandle = user.data.login
+        
+        // Collect incoming data chunks to use in the `on("end")` event
+        let body = '';
+        request.on('data', chunk => {
+            body += chunk.toString();
+        });
+
+        // Parse the collected data once the request ends
+        request.on('end', async () => {
+            try {                
+                // start sending the response back to Copilot
+                console.log('Handling response');
+                // header to indicate the state
+                response.writeHead(200, { 'Content-Type': 'application/json' });
+                // write the acknowledge event to let Copilot know we are handling the request
+                // this will also show the message "Copilot is responding" in the chat
+                response.write(createAckEvent());
+
+                // parse the incoming body as that has the information we need to handle the request / user prompt
+                const payload = parseRequestBody(body);
```

Comment:
I think I would create a separate method to read the body text instead of moving all the core logic into the         `request.on('end', async () => {` handler. And in this example, we should verify the payload using https://github.com/copilot-extensions/preview-sdk.js/?tab=readme-ov-file#async-verifyrequestbykeyidrawbody-signature-keyid-requestoptions